x-pack/solutions/observability/plugins/streams_app/public/components/stream_detail_enrichment/processors/grok/grok_patterns_editor.tsx

```diff
@@ -58,42 +71,56 @@ export const GrokPatternsEditor = () => {
 
   const getRemovePatternHandler = (id: number) => (fields.length > 1 ? () => remove(id) : null);
 
+  const addButton = (
+    <EuiButtonEmpty
+      data-test-subj="streamsAppGrokPatternsEditorAddPatternButton"
+      onClick={handleAddPattern}
+    >
+      {i18n.translate(
+        'xpack.streams.streamDetailView.managementTab.enrichment.processor.grokEditor.addPattern',
+        { defaultMessage: 'Add pattern' }
+      )}
+    </EuiButtonEmpty>
+  );
+
   return (
-    <EuiFormRow
-      label={i18n.translate(
-        'xpack.streams.streamDetailView.managementTab.enrichment.processor.grokEditorLabel',
-        { defaultMessage: 'Grok patterns editor' }
+    <>
+      <EuiFormRow
+        label={i18n.translate(
+          'xpack.streams.streamDetailView.managementTab.enrichment.processor.grokEditorLabel',
+          { defaultMessage: 'Grok patterns editor' }
+        )}
+      >
+        <EuiPanel color="subdued" paddingSize="s">
+          <SortableList onDragItem={handlerPatternDrag}>
+            {fieldsWithError.map((field, idx) => (
+              <DraggablePatternInput
+                key={field.id}
+                field={field}
+                idx={idx}
+                onRemove={getRemovePatternHandler(idx)}
+                inputProps={register(`patterns.${idx}.value`, {
+                  required: i18n.translate(
+                    'xpack.streams.streamDetailView.managementTab.enrichment.processor.grokEditorRequiredError',
+                    { defaultMessage: 'A pattern is required.' }
+                  ),
+                })}
+              />
+            ))}
+          </SortableList>
+        </EuiPanel>
+      </EuiFormRow>
+      {isAiEnabled && refreshSimulation && definition && samples ? (
+        <GrokAiSuggestions
+          definition={definition}
+          refreshSimulation={refreshSimulation}
+          samples={samples}
+          extraButtons={addButton}
+        />
+      ) : (
+        addButton
       )}
```

Comment:
I believe the Ai suggestions here shouldn't be coupled to this component specifically, but added at the GrokProcessorForm component level. This is a component that should purely handle the form patterns value, and doesn't need to know about the samples nor the definition.

Reply:
Here is where we start getting more complex state to manage also suggestions related to the simulation itself the form changes. Introducing a context would work, and the plan was to also manage it with [[Streams] Enrichment - Migrate to solid state management system for stream app #102](https://github.com/elastic/streams-program/issues/102) re-work for maintainability.

For this part now, context looks good, although I'm not 100% convinced about the structure this form is taking.

Reply:
I'll go with context for now, I agree that we anyway want to rework the state management here in a follow-up.